cypress/e2e/invoices/fiscal-year-is-not-etitable-if-invoice-is-payed-or-cancelled.cy.js

```diff
@@ -0,0 +1,116 @@
+    servicePoint: ServicePoints.defaultServicePoint,
+    location: {},
+    user: {},
+  };
+  const status = 'Paid';
+
+  before('Create test data', () => {
+    cy.getAdminToken();
+
+    const { fiscalYear, fund, budget } = Budgets.createBudgetWithFundLedgerAndFYViaApi();
+    testData.budget = budget;
+    testData.fiscalYear = fiscalYear;
+
+    ServicePoints.createViaApi(testData.servicePoint).then(() => {
+      testData.location = Locations.getDefaultLocation({
+        servicePointId: testData.servicePoint.id,
+      });
+
+      Locations.createViaApi(testData.location).then(() => {
+        Organizations.createOrganizationViaApi(testData.organization).then(() => {
+          const orderLine = BasicOrderLine.getDefaultOrderLine({
+            fundDistribution: [{ fundId: fund.id, value: budget.allowableEncumbrance }],
+            spesialLocationId: testData.location.id,
+            vendorAccount: testData.organization.name,
+          });
+
+          Orders.createOrderWithOrderLineViaApi(testData.order, orderLine).then((order) => {
+            Orders.updateOrderViaApi({ ...order, workflowStatus: 'Open' });
+
+            Invoices.createInvoiceWithInvoiceLineViaApi({
+              vendorId: testData.organization.id,
+              poLineId: orderLine.id,
+              fundDistributions: [{ fundId: fund.id, value: budget.allowableEncumbrance }],
+              accountingCode: testData.organization.erpCode,
+            }).then((invoice) => {
+              testData.invoice = invoice;
+
+              Invoices.changeInvoiceStatusViaApi({ invoice, status });
+            });
+          });
+        });
+      });
+    });
+
+    cy.createTempUser([
+      Permissions.uiInvoicesCanViewAndEditInvoicesAndInvoiceLines.gui,
+      Permissions.uiInvoicesPayInvoicesInDifferentFiscalYear.gui,
+    ]).then((userProperties) => {
+      testData.user = userProperties;
+
+      cy.login(userProperties.username, userProperties.password, {
+        path: TopMenu.invoicesPath,
+        waiter: Invoices.waitLoading,
+      });
+    });
+  });
+
+  after('Delete test data', () => {
+    Organizations.deleteOrganizationViaApi(testData.organization.id);
+    ServicePoints.deleteViaApi(testData.servicePoint.id);
+    Users.deleteViaApi(testData.user.userId);
+  });
+
+  it(
+    'C387536 "Fiscal year" field is not editable for paid invoice (thunderjet) (TaaS)',
+    { tags: [TestTypes.criticalPath, DevTeams.thunderjet] },
+    () => {
+      Invoices.searchByNumber(testData.invoice.vendorInvoiceNo);
+      Invoices.selectInvoice(testData.invoice.vendorInvoiceNo);
+      Invoices.checkInvoiceDetails({
+        ...testData.invoice,
+        status,
+        fiscalYear: testData.fiscalYear.code,
+      });
+      const InvoiceEditForm = Invoices.openInvoiceEditForm();
+      InvoiceEditForm.checkButtonsConditions([
+        {
+          label: 'Fiscal year',
+          conditions: { disabled: true, singleValue: testData.fiscalYear.code },
```

Comment:
probably  better use Selection interactor instead of Button. Selection alredy has singleValue filter. No need to add it to Button

Reply:
One of the verifications is to check that button is disabled. Selection interactor doesn't have such filter.